### PR TITLE
Feature: add test mode notice

### PR DESCRIPTION
**Pull request by glaubersilva**
## Description

This PR adds a test mode notice to the Payment Gateways field. This is displayed only when Give is under testing mode.

## Affects

The Payment Gateways field.

## Visuals

image

## Testing Instructions

Just set Give mode to "test" and check if the notice is displayed when the form renders.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205043276379238

### CODE REVIEW

**Comment by glaubersilva** [comment]
@jonwaldstein This is ready for review.

File: src/DonationForms/resources/registrars/templates/fields/Gateways.tsx [hunk 1]
```diff
@@ -20,17 +20,39 @@ function GatewayFieldsErrorFallback({error, resetErrorBoundary}) {
...
+                    fillRule="evenodd"
+                    clipRule="evenodd"
+                    d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM12 7C11.4477 7 11 7.44772 11 8C11 8.55228 11.4477 9 12 9H12.01C12.5623 9 13.01 8.55228 13.01 8C13.01 7.44772 12.5623 7 12.01 7H12ZM13 12C13 11.4477 12.5523 11 12 11C11.4477 11 11 11.4477 11 12V16C11 16.5523 11.4477 17 12 17C12.5523 17 13 16.5523 13 16V12Z"
+                    fill="#F29718"
+                />
+            </svg>{' '}
+            <p>
+                {__('Test mode is', 'give')} <strong>{__('enabled', 'give')}</strong>.{' '}
+                {__('While in test mode no live donations are processed.', 'give')}
+            </p>
```
Comment (jonwaldstein) [inline src/DonationForms/resources/registrars/templates/fields/Gateways.tsx:37]: Let's use [createInterpolateElement](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-element/#createinterpolateelement) here.  We discovered it while building the messaging for the amount field. It allows you to render jsx in your string while preserving the translation

**Review by jonwaldstein — CHANGES_REQUESTED** [review]
@glaubersilva thanks for whipping this up! just a couple minor suggestions - otherwise looking great

File: src/Framework/FieldsAPI/PaymentGateways.php [hunk 2]
```diff
@@ -4,7 +4,23 @@
...
+    public $isTestMode;
+
+    /**
+     * @unreleased
+     */
+    public function setTestMode( bool $testMode ): PaymentGateways {
+        $this->isTestMode = $testMode;
+
+        return $this;
+    }
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/Framework/FieldsAPI/PaymentGateways.php]: Resolved in 1660145b2ac618e5a6d011d97aa5ae7a581b3c67

File: src/DonationForms/resources/registrars/templates/fields/Gateways.tsx (hunk 1 above)
Comment (glaubersilva) [reply src/DonationForms/resources/registrars/templates/fields/Gateways.tsx]: Resolved in 40a9afb70db8e5997bdd920f3128d66970c1ca0f

**Comment by glaubersilva** [comment]
@jonwaldstein Thanks for the suggestions, let me know if there is anything more.

**Review by jonwaldstein — APPROVED** [review]
@glaubersilva great job man!

**Review by jdghinson — APPROVED** [review]
It's all good with the design

**Pull request merged**